twisted/test/proto_helpers.py

```diff
@@ -154,13 +161,14 @@ class StringTransport:
 
     producerState = 'producing'
 
-    def __init__(self, hostAddress=None, peerAddress=None):
+    def __init__(self, hostAddress=None, peerAddress=None, lenient=False):
         self.clear()
         if hostAddress is not None:
             self.hostAddr = hostAddress
         if peerAddress is not None:
             self.peerAddr = peerAddress
         self.connected = True
+        self.lenient = lenient
```

Comment:
This attribute should be private, as per https://glyph.twistedmatrix.com/2015/05/separate-your-fakes-and-your-inspectors.html - we are publicly exposing some things that aren't part of "real" transports here, but we shouldn't be adding more errors of that variety :).